Makefile

```diff
@@ -114,3 +115,11 @@ kind-e2e-tests: build-docker-operator chart setup-kind
 reload-operator: build-docker-operator chart
 	kind load docker-image --name $(CLUSTER_NAME) ${REPO}:${TAG}
 	helm upgrade -n cattle-elemental-system elemental-operator $(CHART)
+
+.PHONY: vendor
+vendor:
+	go mod vendor
+	curl -L 'https://github.com/google/go-tpm-tools/archive/refs/tags/$(GO_TPM_TAG).tar.gz' --output go-tpm-tools.tar.gz
```

Comment:
`go mod vendor` does not include CGO code, hence we need to get source tarball manually and include it. So in order to properly build a functional  `vendor` folder `make vendor` is required.

Reply:
How about about having a git submodule for the go-tpm-tools inside the vendor dir? Would that work or does the go mod vendor overwrite it?

I know we can replace the module in the go.mod to point to a local sir, so that would avoid having to download the tar on the makefile which is a bit yuck

Reply:
Damn, this crap has been going on for 4 years! https://github.com/golang/go/issues/26366

There is a couple of simple fixes but those need to go upstream.. I will check tonigth to see if there is a saner way to get this proper

Reply:
> Damn, this crap has been going on for 4 years! [golang/go#26366](https://github.com/golang/go/issues/26366)

lol, I've there too 😅

Reply:
🤦🏻‍♂️